ble/plf/hci_uart.c

```diff
@@ -225,15 +230,68 @@ void hci_uart_callback(const struct device *dev, void *user_data)
 	/* TODO error/overflow handling */
 }
 
+static bool hci_uart_rx_dma_driver_check(void)
+{
+/* RX DMA is disabled until we find a proper solution */
+#if CONFIG_UART_ASYNC_API && HCI_RX_DMA_ENABLED
+
+	/* RX DMA is disabled because it not give great value yet */
+#if DT_NODE_HAS_STATUS(DT_DMAS_CTLR_BY_NAME(DT_NODELABEL(uart_hci), rx), okay)
+	const struct device *rxdma =
+		DEVICE_DT_GET_OR_NULL(DT_DMAS_CTLR_BY_NAME(DT_NODELABEL(uart_hci), rx));
+
+	if (rxdma) {
+
+		if (device_is_ready(rxdma)) {
+			int rx_config = DT_DMAS_CELL_BY_NAME(DT_NODELABEL(uart_hci), rx, periph);
+
+			LOG_DBG("DMA RX event enable %d", rx_config);
+			dma_event_router_configure(DMA_UART_RX_GROUP, rx_config, false);
+			uart_callback_set(uart_dev, hci_uart_async_callback, NULL);
+
+			return true;
+		}
+	}
+#else
+	LOG_ERR("NO RX DMA");
+#endif
+#endif
+	uart_irq_rx_enable(uart_dev);
+	uart_irq_callback_user_data_set(uart_dev, hci_uart_callback, NULL);
+	return false;
+}
+
+static bool hci_uart_tx_dma_driver_check(void)
+{
+#ifdef CONFIG_UART_ASYNC_API
+#if DT_NODE_HAS_STATUS(DT_DMAS_CTLR_BY_NAME(DT_NODELABEL(uart_hci), rx), okay)
+	const struct device *txdma =
+		DEVICE_DT_GET_OR_NULL(DT_DMAS_CTLR_BY_NAME(DT_NODELABEL(uart_hci), tx));
+
+	if (txdma) {
+
+		if (device_is_ready(txdma)) {
+			int tx_config = DT_DMAS_CELL_BY_NAME(DT_NODELABEL(uart_hci), tx, periph);
+			LOG_DBG("DMA TX event enable %d", tx_config);
```

Comment:
You declare a variable on one line and have code right on the next one, this doesn't generate a checkpatch warning?